samples/matter/lock/src/flash-external.py

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env python3
+#
+# Copyright (c) 2022 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: Apache-2.0
```

Comment:
We might want to update the https://github.com/nrfconnect/sdk-nrf/blob/main/scripts/ci/license_allow_list.yaml